tests/foreman/api/test_ansible.py

```diff
@@ -253,3 +255,61 @@ def test_add_and_remove_ansible_role_hostgroup(target_sat):
         target_sat.api.HostGroup(id=hg.id).remove_ansible_role(data={'ansible_role_id': role})
     host_roles = target_sat.api.HostGroup(id=hg.id).list_ansible_roles()
     assert len(host_roles) == 0
+
+
+@pytest.mark.rhel_ver_match('[7]')
+@pytest.mark.tier2
+def test_positive_read_facts_with_filter(target_sat, module_org, module_location, rex_contenthost):
+    """
+    Read host's Ansible facts as a user with a role that has host filter
+
+    :id: 483d5faf-7a4c-4cb7-b14f-369768ad99b0
+
+    :Steps:
+        1. Create a role with a host view filtered
+        2. Create a user with that role
+        3. Create a host
+        4. Run Ansible roles on a host
+        5. Using API, read Ansible facts of that host
+
+    :expectedresults: Ansible facts returned
+
+    :BZ: 1699188
+
+    :customerscenario: true
+    """
+    api = target_sat.api
+
+    role = api.Role(
+        name=gen_string('alpha'), location=[module_location], organization=[module_org]
+    ).create()
+    # assign view_hosts (with a filter, to test BZ 1699188),
+    # view_hostgroups, view_facts permissions to the role
+    permission_hosts = api.Permission().search(query={'search': 'name="view_hosts"'})
+    permission_hostgroups = api.Permission().search(query={'search': 'name="view_hostgroups"'})
+    permission_facts = api.Permission().search(query={'search': 'name="view_facts"'})
+    api.Filter(permission=permission_hosts, search='name != nonexistent', role=role).create()
+    api.Filter(permission=permission_hostgroups, role=role).create()
+    api.Filter(permission=permission_facts, role=role).create()
+
+    password = gen_string('alpha')
+    user = api.User(
+        role=[role], password=password, location=[module_location], organization=[module_org]
+    ).create()
+
+    host = api.Host().search(query={'search': f'name={rex_contenthost.hostname}'})[0]
+    host_id = host.id
+    api.Host(id=host_id, organization=[module_org.id]).update(['organization'])
+    api.Host(id=host_id, location=module_location.id).update(['location'])
+    # gather ansible facts by running ansible roles on the host
+    host.play_roles()
+
+    sc = ServerConfig(auth=(user.login, password), url=target_sat.url, verify=False)
+    host = entities.Host(sc).search(query={'search': f'name={rex_contenthost.hostname}'})[0]
```

Comment:
```suggestion
    host = api.Host(sc).search(query={'search': f'name={rex_contenthost.hostname}'})[0]
```

Reply:
No:
```
FAILED tests/foreman/api/test_ansible.py::test_positive_read_facts_with_filter[rhel7] - TypeError: __init__() got multiple values for argument 'server_config'
```
Notice that I use a different ServerConfig than default. That is because I need to use a different user - the one I just created. Is there any better way to do it, @shweta83?

Reply:
Missed ServerConfig there. Your approach is fine.